mod_form.php

```diff
@@ -55,33 +55,80 @@ public function definition() {
+            $firstoption = empty($firstoption) ? 'verifyany' : $firstoption;
+        } else {
+            $mform->addElement('hidden', 'verifyany', get_config('customcert', 'verifyany'));
+        }
+        $mform->setType('verifyany', PARAM_INT);
 
-        $mform->addElement('text', 'requiredtime', get_string('coursetimereq', 'customcert'), array('size' => '3'));
+        if (has_capability('mod/customcert:managerequiredtime', $this->get_context())) {
+            $mform->addElement('text', 'requiredtime', get_string('coursetimereq', 'customcert'), array('size' => '3'));
+            $mform->addHelpButton('requiredtime', 'coursetimereq', 'customcert');
+            $mform->setDefault('requiredtime', get_config('customcert', 'coursetimereq'));
+            $firstoption = empty($firstoption) ? 'requiredtime' : $firstoption;
+        } else {
+            $mform->addElement('hidden', 'requiredtime', 0);
+        }
         $mform->setType('requiredtime', PARAM_INT);
-        $mform->addHelpButton('requiredtime', 'coursetimereq', 'customcert');
 
-        $mform->addElement('checkbox', 'protection_print', get_string('setprotection', 'customcert'),
-            get_string('print', 'customcert'));
-        $mform->addElement('checkbox', 'protection_modify', '', get_string('modify', 'customcert'));
-        $mform->addElement('checkbox', 'protection_copy', '', get_string('copy', 'customcert'));
-        $mform->addHelpButton('protection_print', 'setprotection', 'customcert');
+        if (has_capability('mod/customcert:manageprotection', $this->get_context())) {
+            $mform->addElement('checkbox', 'protection_print', get_string('setprotection', 'customcert'),
+                get_string('print', 'customcert'));
+            $mform->addElement('checkbox', 'protection_modify', '', get_string('modify', 'customcert'));
+            $mform->addElement('checkbox', 'protection_copy', '', get_string('copy', 'customcert'));
+            $mform->addHelpButton('protection_print', 'setprotection', 'customcert');
+            $mform->setDefault('protection_print', get_config('customcert', 'protection_print'));
+            $mform->setDefault('protection_modify', get_config('customcert', 'protection_modify'));
+            $mform->setDefault('protection_copy', get_config('customcert', 'protection_copy'));
+            $firstoption = empty($firstoption) ? 'protection_print' : $firstoption;
+        } else {
+            $mform->addElement('hidden', 'protection_print', get_config('customcert', 'protection_print'));
+            $mform->addElement('hidden', 'protection_modify', get_config('customcert', 'protection_modify'));
+            $mform->addElement('hidden', 'protection_copy', get_config('customcert', 'protection_copy'));
+        }
+        $mform->setType('protection_print', PARAM_INT);
+        $mform->setType('protection_modify', PARAM_INT);
+        $mform->setType('protection_copy', PARAM_INT);
```

Comment:
It probably won't make a huge difference, but would it make sense to change these three lines from PARAM_INT to PARAM_BOOL?

Reply:
Yes, this sounds sense. will change it